mods/beds/beds.lua

```diff
@@ -102,3 +140,28 @@ minetest.register_craft({
 	recipe = "beds:bed_bottom",
 	burntime = 12,
 })
+
+-- colored Crafting
+
+minetest.register_on_craft(function(itemstack, player, old_craft_grid, craft_inv)
```

Comment:
You don't need a server-side function to do the craft colorization if you only support the default dyes: you can register a craft recipe for each dye then. In lua_api.txt you can find an example how to make colored craft recipes in the Hardware coloring section.